exercises/robot-simulator/robot-simulator.test.ts

```diff
@@ -0,0 +1,153 @@
+      expect(exception).toEqual('Invalid Robot Bearing')
+    }
+  })
+
+  xit('turn right from north', () => {
+    robot.orient('north')
+    robot.turnRight()
+    expect(robot.bearing).toEqual('east')
+  })
+
+  xit('turn right from east', () => {
+    robot.orient('east')
+    robot.turnRight()
+    expect(robot.bearing).toEqual('south')
+  })
+
+  xit('turn right from south', () => {
+    robot.orient('south')
+    robot.turnRight()
+    expect(robot.bearing).toEqual('west')
+  })
+
+  xit('turn right from west', () => {
+    robot.orient('west')
+    robot.turnRight()
+    expect(robot.bearing).toEqual('north')
+  })
+
+  xit('turn left from north', () => {
+    robot.orient('north')
+    robot.turnLeft()
+    expect(robot.bearing).toEqual('west')
+  })
+
+  xit('turn left from east', () => {
+    robot.orient('east')
+    robot.turnLeft()
+    expect(robot.bearing).toEqual('north')
+  })
+
+  xit('turn left from south', () => {
+    robot.orient('south')
+    robot.turnLeft()
+    expect(robot.bearing).toEqual('east')
+  })
+
+  xit('turn left from west', () => {
+    robot.orient('west')
+    robot.turnLeft()
+    expect(robot.bearing).toEqual('south')
+  })
+
+  xit('robot coordinates', () => {
+    robot.at(3, 0)
+    expect(robot.coordinates).toEqual([3, 0])
+  })
+
+  xit('other robot coordinates', () => {
+    robot.at(-2, 5)
+    expect(robot.coordinates).toEqual([-2, 5])
+  })
+
+  xit('advance when facing north', () => {
+    robot.at(0, 0)
+    robot.orient('north')
+    robot.advance()
+    expect(robot.coordinates).toEqual([0, 1])
+  })
+
+  xit('advance when facing east', () => {
+    robot.at(0, 0)
+    robot.orient('east')
+    robot.advance()
+    expect(robot.coordinates).toEqual([1, 0])
+  })
+
+  xit('advance when facing south', () => {
+    robot.at(0, 0)
+    robot.orient('south')
+    robot.advance()
+    expect(robot.coordinates).toEqual([0, -1])
+  })
+
+  xit('advance when facing west', () => {
+    robot.at(0, 0)
+    robot.orient('west')
+    robot.advance()
+    expect(robot.coordinates).toEqual([-1, 0])
+  })
+
+  xit('instructions for turning left', () => {
+    expect(robot.instructions('L')).toEqual(['turnLeft'])
+  })
+
+  xit('instructions for turning right', () => {
+    expect(robot.instructions('R')).toEqual(['turnRight'])
+  })
+
+  xit('instructions for advancing', () => {
+    expect(robot.instructions('A')).toEqual(['advance'])
+  })
+
+  xit('series of instructions', () => {
+    expect(robot.instructions('RAAL'))
+      .toEqual(['turnRight', 'advance', 'advance', 'turnLeft'])
+  })
+
+  xit('instruct robot', () => {
+    robot.place({ x: -2, y: 1, direction: 'east' })
+    robot.evaluate('RLAALAL')
+    expect(robot.coordinates).toEqual([0, 2])
+    expect(robot.bearing).toEqual('west')
+  })
+
+  xit('instruct many robots', () => {
+    const robot1 = new Robot()
+    const robot2 = new Robot()
+    const robot3 = new Robot()
+    robot1.place({ x: 0, y: 0, direction: 'north' })
```

Comment:
This looks like a contractor. What do you think about changing into a constructor but using explicit arguments?